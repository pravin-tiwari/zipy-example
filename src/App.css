* {
  margin: 0;
  padding: 0;
  font-family: 'Gill Sans', 'Gill Sans MT', Calibri, 'Trebuchet MS', sans-serif;
  font-size: medium;
}

body {
  background-color: lightgray;
}

nav {
  height: 50px;
  margin: 6px 10%;
  line-height: 3;
  text-align: center;
}

nav ul {
  display: flex;
  flex-direction: row;
}

nav ul li {
  flex: 1;
  text-decoration: none;
  list-style: none;
}

nav ul li a {
  text-decoration: none;
  color: black;
  font-weight: lighter;
}

nav ul li a.active {
  font-weight: 600;
}