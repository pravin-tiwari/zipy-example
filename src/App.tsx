
import './App.css';
import {
  BrowserRouter as Router,
  Routes,
  Route,
  NavLink
} from "react-router-dom";
import Dashboard from './components/Dashboard/Dashboard.lazy';
import Account from './components/Account/Account.lazy';
import Home from './components/Home/Home.lazy';

function App() {
  return (
    <Router>
      <div data-testid="App">
        <nav>
          <ul>
            <li>
              <NavLink to="/" >Dashboard</NavLink>
            </li>
            <li>
              <NavLink to="/home" >Home</NavLink>
            </li>
            <li>
              <NavLink to="/account" >Account</NavLink>
            </li>
          </ul>
        </nav>
        <Routes>
          <Route path="/home" element={<Home />} />
          <Route path="/account" element={<Account />} />
          <Route path="/" element={<Dashboard />} />
        </Routes>
      </div>
    </Router>
  );
}

export default App;
