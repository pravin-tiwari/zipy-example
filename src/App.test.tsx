import { render, waitFor } from '@testing-library/react';
import App from './App';

describe('<App />', () => {
  test('it should mount', async () => {
    const { getByTestId } = render(<App />);
    const app = await waitFor(() => getByTestId('App'));
    expect(app).toBeInTheDocument();
  });
});