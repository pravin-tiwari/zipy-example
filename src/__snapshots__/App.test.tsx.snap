// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> it should mount 1`] = `
<div
  class="App"
  data-testid="app-id"
/>
`;
